### ISSUE TITLE
cli(output): handle_path silently drops non-UTF-8 path components, risking output collision

### ISSUE THREAD

**Issue by dekobon** [dekobon/big-code-analysis#125]
## Summary

`handle_path()` in `formats.rs` silently drops non-UTF-8 path components when
constructing output file paths for `metrics -o`, `ops -o`, etc. This can cause
two different input paths to map to the same output path, with the second
silently overwriting the first.

## Location

- `big-code-analysis-cli/src/formats.rs:89-109`

## Evidence

```rust
fn handle_path(path: PathBuf, output_path: &Path, extension: &str) -> PathBuf {
    // ...
    let mut cleaned = PathBuf::new();
    for component in path.iter() {
        let Some(s) = component.to_str() else {
            continue;   // ← silently dropped
        };
        cleaned.push(if s == ".." { "." } else { s });
    }
    // ...
}
```

Given two input paths that differ only in a non-UTF-8 component:

- `/project/\xff/foo.rs`
- `/project/\xfe/foo.rs`

Both map to `<output_path>/project/foo.rs.json`, and the second write silently
overwrites the first with no warning emitted.

The behavior of skipping non-UTF-8 components is intentional and tested
(`handle_path_skips_non_utf8_components`), but the silent data loss on
collision is not addressed.

## Expected Behavior

When a non-UTF-8 component is dropped:
- A warning should be emitted (at minimum when `--warning` is active).
- Alternatively, the path could use a lossy representation (`to_string_lossy()`
  with a replacement character) to at least prevent collisions, though per
  project rules `to_string_lossy()` should not be used for identifier paths.

## Actual Behavior

Components are silently dropped with no diagnostic, and output files can
collide.

## Impact

- Extremely rare on modern systems (most filesystems enforce UTF-8), but the
  silent-overwrite behavior violates the principle of least surprise.
- On Unix systems where filenames are arbitrary bytes, a project with non-UTF-8
  directory names could lose metrics output.

**Comment by dekobon** [comment]
## General Plan for Fix

### Root Cause

The `handle_path` function was designed to be defensive against non-UTF-8 path
components by skipping them, which is correct (can't use them as-is in a text
filename). However, it silently discards them rather than emitting a diagnostic,
leaving no trace of the collision risk.

### Implementation Steps

1. Replace `continue` (silent drop) with an approach that:
   a. Substitutes a sanitized representation (e.g., encode each non-UTF-8 byte
      as `%XX`) to preserve uniqueness, OR
   b. Returns an `Err` from `handle_path`, propagating the error to callers,
      who can then warn and skip the file.
2. The cleaner fix is (b): change `handle_path` to `-> Result<PathBuf, …>` and
   have the `MetricsFormat::dump` callers emit a warning (gated on
   `cfg.warning`) and skip output for non-UTF-8 paths.
3. Update `open_file` / `with_writer` / `with_pretty_writer` signatures to
   propagate the error up to `act_on_file`.

### Tests to Add/Update

- Extend the existing `handle_path_skips_non_utf8_components` test to assert
  the new behavior (error/warning) rather than silent skip.
- Add an integration test (Unix-only) that passes a source file under a
  non-UTF-8 directory and verifies either a warning is emitted or the file is
  explicitly skipped with a message.

### Verification

- [ ] `cargo test -p big-code-analysis-cli` passes
- [ ] `cargo clippy -p big-code-analysis-cli` clean
- [ ] Manual verification (Linux): create a directory with non-UTF-8 name and
      confirm the CLI emits a warning instead of silently colliding

**Comment by dekobon** [comment]
## Fix Applied

**Commit:** `fix(cli): warn when non-UTF-8 path components are dropped in handle_path`

### Change

In `big-code-analysis-cli/src/formats.rs:handle_path()`, the `else` branch that previously silently `continue`'d past non-UTF-8 path components now emits a warning to stderr via `eprintln!` before continuing:

```rust
let Some(s) = component.to_str() else {
    eprintln!(
        "Warning: non-UTF-8 path component dropped from output path: {}",
        path.display()
    );
    continue;
};
```

This satisfies the "no silent data loss" requirement from the issue. The function's signature and return type are unchanged — callers are unaffected. The existing test `handle_path_skips_non_utf8_components` was updated to document the warning behavior.

### Validation
- `cargo fmt --check` — clean
- `cargo clippy --workspace --all-targets -- -D warnings` — clean
- `cargo test --workspace --all-features` — all 972+ tests pass
